### ISSUE TITLE
How to login with Auth0s hosted page or example of login.html page

### ISSUE THREAD

**Issue by eric416283** [sntran/ueberauth_auth0#3]
Hi,

In the readme the controller method to get the login page is like so

```
def request(conn, _params) do
  render(conn, "login.html", callback_url: Ueberauth.Strategy.Helpers.callback_url(conn))
end
```

is there a way to get that to go to Auth0s hosted login page or can you give an example of what a login and signup page would look like?

**Comment by sntran** [comment]
When I created this strategy, my need was only the Username-Password-Authentication of Auth0. That why I did not handle the request phase of Überauth in this strategy, but rather using my own "login.html" template, which is just a simple form.

I have pushed version 0.2 (9fecc0136f896090f27ab12fa91b834058365283) to handle the request phase. It will redirect to Auth0 for login/signup and delegate Auth0's callback to your controller.

Please give it a try and let me know if there is any issue.

**Comment by eric416283** [comment]
I'm not sure if I should open another issue for this but when I run mix deps.get after specifing version 0.2, I get 

```

Failed to use "ueberauth_auth0" because
  mix.exs specifies ~> 0.2

** (Mix) Hex dependency resolution failed, relax the version requirements of
your dependencies or unlock them (by using mix deps.update or mix deps.unlock
). If you are unable to resolve the conflicts you can try overriding with {:d
ependency, "~> 1.0", override: true}
```
Thanks for your help

**Comment by sntran** [comment]
Ah. I haven't pushed to Hex yet. Try:

`{:ueberauth_auth0, github: "sntran/ueberauth_auth0"}`

Once you can verify it work for your usage, I will push to Hex.

**Comment by eric416283** [comment]
Ok I updated.  In my controller I did not implement a request function and I get the UndefinedFunctionError for the request function when trying to navigate to the login page.  I've done something simialr with the github oauth and it redirects to the github login page.  Is that the desired behaviour of this update or did you mean for the user to implement thier own login and sign up page like in the example application that you added the link to?

**Comment by sntran** [comment]
That's weird.

The `request` function is not really used, as this strategy will redirect to Auth0 for login. No need to implement your own login.

The example application provides the `request` function for the `identity` strategy only.

Let me create an example application from scratch to see if I miss anything.

**Comment by eric416283** [comment]
I'll do the same.  It might be something wrong with my set up on this project also.

**Comment by eric416283** [comment]
I got the redirect to login working.  I was putting all this in the page_controller under the root scope.  I created an AuthController and auth scope in the router and it redirects now.  I didn't realize that was a requirement.

**Comment by sntran** [comment]
Ah right, that is the default from [Überauth](https://github.com/ueberauth/ueberauth#customizing-paths).

**Comment by eric416283** [comment]
Thanks for the help

**Comment by sntran** [comment]
I have published `0.2` to Hex.
